backend/server/index.js

```diff
@@ -18,18 +18,42 @@ const io = require('socket.io')(http, {
   },
 });
 
+const clients = {};
+const clientsById = {};
+
 io.on('connection', (socket) => {
   console.log(`⚡: ${socket.id} user just connected!`);
-  socket.join('room1');
-  socket.on('requestID', () => {
-    socket.emit('sendID', socket.id);
+  socket.join(socket.id);
+  socket.on('requestID', (arg) => {
+    let newUser = { id: arg, socket_id: socket.id };
+    let secondUser = false;
+    Object.keys(clients).forEach((key) => {
+      if (clients[key] === 87) {
+        secondUser = true;
+      }
+    });
+    if (secondUser) {
+      newUser = { id: 34, socket_id: socket.id };
+      clients[socket.id] = newUser.id;
+      clientsById[newUser.id] = socket.id;
+      socket.emit('sendID', newUser);
+    } else {
+      clients[socket.id] = newUser.id;
+      clientsById[newUser.id] = socket.id;
+      socket.emit('sendID', newUser);
+    }
   });
   socket.on('send', (arg) => {
-    //console.log(arg);
-    io.to('room1').emit('response', arg);
+    console.log(arg);
+    const recipientRoom = clientsById[JSON.stringify(arg.recipient_id)];
+    const senderRoom = clientsById[JSON.stringify(arg.sender_id)];
+    io.to(recipientRoom).emit('response', arg);
+    io.to(senderRoom).emit('response', arg);
   });
   socket.on('disconnect', () => {
+    delete clients[socket.id];
     console.log('🔥: A user disconnected');
+    console.log('now clients are ', clients);
```

Comment:
Is this produciton-grade? If so, please delete console logs

Reply:
This one is not necessary for now. Will remove.